content/en/blog/_posts/2024-07-18-kubernetes-1.31-deprecations-and-removals.md

```diff
@@ -0,0 +1,116 @@
+
+## A note about SHA-1 signature support fully going away in go1.24
+
+In [go1.18](https://go.dev/doc/go1.18#sha1) (released in March 2022), the crypto/x509 library started to reject certificates that were signed with a SHA-1 hash function. While SHA-1 is established to be unsafe and publicly trusted Certificate Authorities have not issues SHA-1 certificates since 2015, there might still be cases in the context of Kubernetes where user-provided certificates are signed using a SHA-1 hash function through private authorities with them being used for Aggregated API Servers of webhooks. If you have been reliant on SHA-1 based certificates, you must be explicitly opting back into its support by setting `GODEBUG=x509sha1=1` in your environment.
+
+Given Go's [compatibility policy for GODEBUGs](https://go.dev/blog/compat), the `x509sha1` GODEBUG and consequently the support for SHA-1 certificates will [fully go away in go1.24](https://tip.golang.org/doc/go1.23) which would release in the first half of 2025. If you are reliant on SHA-1 certificates, please start moving off of them.
+
+Please see [Kubernetes issue #125689](https://github.com/kubernetes/kubernetes/issues/125689) to get a better idea of timelines around the support for SHA-1 going away, when Kubernetes releases plans to adopt go1.24, and for more details on how to detect usage of SHA-1 certificates via metrics and audit logging. 
+
+## Deprecations and Removals in Kubernetes 1.31
+
+### Deprecate kustomize from kubectl [KEP 4706](https://github.com/kubernetes/enhancements/issues/4706)
+In the early days of Kubernetes, when not that many tools in the field of declarative configurations existed, [kustomize](https://github.com/kubernetes-sigs/kustomize) was introduced and to make life easier for kubectl users it was included in kubectl, by default.
+
+Over the past several years the ecosystem around Kubernetes grew significantly, resulting in wider spread of tooling, including declarative configuration. With these new options, it's best left to the user to match their use cases with the best tool capable of resolving the problem at hand. Using [kubectl plugins](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/), users can pick and choose the best tool to match their needs.
+
+With Kubernetes v1.31, kustomize will be deprecated from kubectl. It will be removed in a future release. This will allow both tools to be developed and maintained separately.
+
+### Deprecate Kubelet RunOnce mode [KEP 4580](https://github.com/kubernetes/enhancements/issues/4580)
+The kubelet support for RunOnce mode is being deprecated and will be removed in a future release. Users are advised to deprecate RunOnce mode by disabling the `LegacyNodeRunOnceMode` feature gate. Starting in Kubernetes 1.31, kubelet will refuse to start if this feature gate is disabled. 
```

Comment:
I can not find the feature-gate in the k/k codebase. It seems that the KEP is not implemented yet.

cc @HirazawaUi

If you don't have bandwidth to implement this, please let me know. I'm happy to help.

Reply:
I have submitted a PR, but it has not been merged yet.

Reply:
@HirazawaUi will this be merged before the blog is published? We are planning to publish just want to make sure that this is updated with what get merged in case there are any changes